extension/docs/WebComponentsIntegrationGuide.md

```diff
@@ -0,0 +1,665 @@
+      "type": "Authorization",
+      "amount": {
+        "type": "centPrecision",
+        "currencyCode": "EUR",
+        "centAmount": 1000,
+        "fractionDigits": 2
+      },
+      "interactionId": "853592567856061C",
+      "state": "Success"
+    }
+  ]
+}
+```
+
+## Step 6: Capture payment (required for Klarna)
+All Klarna payments [have to be manually captured](https://docs.adyen.com/payment-methods/klarna/web-component#capture) within 28 days after authorisation, even if you have enabled automatic capture on your merchant account.
+Refer to [Manual Capture](ManualCapture.md) guide to see how it can done.
+
+## Error handling
+
+In case you encounter errors in your integration, refer to the following:
+
+### Extension module errors
+
+If you receive a `non-HTTP 200 response`, use the commercetools `interface interactions` to troubleshoot the request and errors.
+
+Interface interactions can be requests sent to the Adyen, responses received from the Adyen or notifications received from the Adyen. 
+Some interactions may result in a transaction. If so, the interactionId in the Transaction should be set to match the `pspReference` of the Adyen for the interaction.
+
+### Adyen payment refusals
+
+If you receive an HTTP 200 response with an Error or Refused resultCode, check the refusal reason and, if possible, modify your request.
+
+Check the following table to see the mapping of Adyen [result codes](https://docs.adyen.com/development-resources/response-handling#error-codes-types) to commercetools [transaction state](https://docs.commercetools.com/http-api-projects-payments#transactionstate)
+|Adyen result code| The commercetools transaction state
+| --- | --- |
+| Authorised| Success|
+| Refused| Failure|
+| Error| Failure|
+
+### Shopper successfully paid but `redirectUrl` was not reached
+In some payment redirect cases, there might be a valid payment but no order as shopper did not reach the shop's `redirectUrl`.
+For example, after successfully issued payment shopper loses internet connection or accidentally closes the tab.
+In this case [Notification module](../../notification) will receive later a notification with successful content, process it, and update the payment.
+Usage of scheduled [commercetools-payment-to-order-processor](https://github.com/commercetools/commercetools-payment-to-order-processor) job ensures that for every successful payment
+an order can still be asynchronously created.
+
+### Shopper tries to pay a different amount than the actual order amount
+For redirect payments payment amount is bound to `redirectUrl`.
+After redirect and before the actual finalization of the payment at the provider's page, the shopper is still able to change the cart's amount within the second tab.
+If shopper decides to change cart's amount within the second tab and finalize payment within the first tab, then according to payment amount validation an error
+will be shown and order creation must be declined.
```

Comment:
```suggestion
will be shown and order creation must be declined. In such a case, it might be reasonable to [cancel or refund](#cancel-or-refund) the invalid payment.
```